src/simplewallet/simplewallet.cpp

```diff
@@ -3214,6 +3215,54 @@ bool simple_wallet::apropos(const std::vector<std::string> &args)
   return true;
 }
 
+bool simple_wallet::scan_tx(const std::vector<std::string> &args)
+{
+  if (args.empty())
+  {
+    PRINT_USAGE(USAGE_SCAN_TX);
+    return true;
+  }
+
+  // Dedup args
+  std::set<std::string> txids_s;
+  for (const auto &s : args) {
+    if (!string_tools::validate_hex(64, s)) {
+      fail_msg_writer() << tr("Invalid txid specified: ") << s;
+      return true;
+    }
+    txids_s.insert(boost::algorithm::to_lower_copy(s));
+  }
+
+  // Convert txids from string to crypto::hash
+  std::vector<crypto::hash> txids;
+  for (const auto &s : txids_s) {
+    cryptonote::blobdata txid_data;
+    if (!epee::string_tools::parse_hexstr_to_binbuff(s, txid_data) || txid_data.size() != sizeof(crypto::hash)) {
+      fail_msg_writer() << tr("Invalid txid specified: ") << s;
+      return true;
+    }
+    txids.push_back(*reinterpret_cast<const crypto::hash*>(txid_data.data()));
+  }
```

Comment:
That's OK, but there's simpler:

   std::unordered_set<crypto::hash> txids;
   for (const auto &s : args)
  {
    crypto::hash txid;
    if (!epee::string_tools::hex_to_pod(s, txid))
    {
        fail_msg_writer() << tr("Invalid txid specified: ") << s;
        return true;
    }
    txids.insert(txid);  // could also check return value second to warn on dedup
  }